### ISSUE TITLE
Nondeterministic ncread in Julia0.5

### ISSUE THREAD

**Issue by andrewdalecramer** [meggart/NetCDF.jl#33]
I'm trying to read an ExodusII file with code I'd written for Julia0.4.x using the NetCDF package. Using ncread(filename,"coor_names") gives garbage values

I can reproduce the bug with the following.

$ wget https://github.com/idaholab/moose/raw/devel/examples/ex01_inputfile/mug.e
$ julia
julia> f = "mug.e"
julia> using NetCDF
julia> A = ncread(f,"coor_names")
julia> sum([ncread(f,"coor_names")==A for i in 1:1000])

Downloaded julia binary from http://julialang.org/downloads/ for linux 64bit. Was no issue in previous version of julia (can't remember exactly which).

**Comment by visr** [comment]
I see the same on master.
Relevant information:

```
    len_string = 33 ;
    num_dim = 3 ;
char coor_names(num_dim, len_string) ;
 coor_names =
  "x",
  "y",
  "z" ;
```

And `ncread(f,"coor_names")` returns a different 33×3 `Array{UInt8,2}` on every read.

**Comment by meggart** [comment]
Just a short update, the values are actually never read, after merging https://github.com/JuliaGeo/NetCDF.jl/pull/32 one would see the error code -56 (Conversion from text to numbers). 
There is an inconsistency in our interface. The main problem is that we are using Julia dispatch to determine which `nc_get_vara_x` function to use based on the return array. 

This works fine for number types, but there is an ambiguity if a return type of `UInt8` should call `nc_get_vara_text` or `nc_get_vara_ubyte`. The main complication is that some older NetCDF files use NC_CHAR as a numeric format and are supposed to be interpreted as `UInt8`. This has caused several issues before. I will try to fix this tomorrow, but might have to think a bit more how to make a clean interface and document it so that the user can decide if a CHAR/BYTE value is interpreted as text or number.

**Comment by meggart** [comment]
I made a PR which should fix this #34
